packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -64,17 +87,73 @@ export default async (req, res) => {
   })
 }
 
-export class StoreGetProductsParams {
-  @IsInt()
+export class StoreGetProductsPaginationParams {
+  @IsNumber()
+  @IsOptional()
   @Type(() => Number)
-  limit = 100
+  offset?: number = 0
 
-  @IsInt()
+  @IsNumber()
+  @IsOptional()
   @Type(() => Number)
-  offset = 0
+  limit?: number = 100
+}
+
+export class StoreGetProductsParams extends StoreGetProductsPaginationParams {
+  @IsString()
+  @IsOptional()
+  id?: string
+
+  @IsString()
+  @IsOptional()
+  q?: string
 
+  @IsArray()
   @IsOptional()
+  collection_id?: string[]
+
+  @IsOptional()
+  @IsString({ each: true })
+  @IsArray()
+  status = ["published"]
```

Comment:
We don't allow filtering by status. From a storefront, one should always only receive published products, and this is also enforced in the endpoint, so we can safely remove this.